docs/config.toml

```diff
@@ -0,0 +1,55 @@
+# This file contains the config settings available to jj/Jujutsu.
+# It is also a valid TOML file and thus can be used as a template.
```

Comment:
> I'm really fine with either way but I chose this to avoid having two sets of docs for one item. Maybe a template should be ultra minimal? All ways seem equally fair.

I don't think a template needs to be minimal, but I think it will quickly grow too large if it contains all possible config keys. It makes sense to me to have a template for some typical set of keys, for which we can't just make it the default (such as username and email). As an example, I looked at my own `~/.gitconfig` and following seem reasonable to include in a template (for the Git project):
```
[user]
        email = # Fill in
        name = # Fill in
[core]
#        excludesFile = ~/.gitignore
[color]
        ui = true
[alias]
        st = status
        br = branch
        ci = commit
        co = checkout
[branch]
        autosetupmerge = always
        autosetuprebase = always
[merge]
        conflictstyle = diff3
[rebase]
        autosquash = true
```

> If you resolve the other fixes first so I don't miss something, I'll then convert this to markdown.

Thanks, that's now done.

Reply:
I've updated the text for Markdown and tried to keep in mind newer users who may be unfamiliar with VCS or TOML. If you think the changes are okay I can squash/rebase this to clean up the commits (or perhaps you can? I'm not so familiar with Github's controls)

Just let me know.

On the template, sounds good. I'll add that in a separate pull, not sure where it should live though.

Reply:
Thanks for all your work on this!

> I've updated the text for Markdown and tried to keep in mind newer users who may be unfamiliar with VCS or TOML. If you think the changes are okay I can squash/rebase this to clean up the commits (or perhaps you can? I'm not so familiar with Github's controls)
> 
> Just let me know.

I have disabled squash merges, so please do that.

> On the template, sounds good. I'll add that in a separate pull, not sure where it should live though.

Seems fine to put it in `docs/` even though it's not a markdown file. Maybe `docs/config-template.toml` or something like that?

Reply:
I know I wrote I'd put it in a separate pull but I forgot, I'll sort that out later in the rebase 😊 I put in a couple of alternatives in the comments but it's basically bare bones. I checked it parses okay.

> Thanks for all your work on this!

I think we're all grateful for the huge amount of work you've put in. As such, it's my pleasure.